### PR TITLE
Fix food heal overrides

### PR DESCRIPTION
**Pull request by thegrb93**
[BUG]

## About the PR 
Fixing certain food not working

## Why's this needed? 
Some cleanup and bugfixes.

## Changelog

```changelog
(u)thegrb93
(+)Terrible food that gives you food poisoning no longer fills hunger
```

### CODE REVIEW

**Comment by thegrb93** [comment]
Using SpacemanDMM to find all the spots that need fixed.

**Comment by thegrb93** [comment]
Using SpacemanDMM to find all the spots that need fixed

**Comment by thegrb93** [comment]
I couldn't reproduce https://github.com/goonstation/goonstation/issues/11477 locally so I don't think this solves that issue. I know that issue has happened to me too though.

File: code/modules/chemistry/tools/food_and_drink.dm [hunk 1]
```diff
@@ -33,29 +33,30 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food)
...
 			if (prob(20))
 				M.contract_disease(/datum/ailment/disease/food_poisoning, null, null, 1) // path, name, strain, bypass resist
 			healing = 0
 
+		if (healing > 0)
+			if (ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if (H.sims)
+					H.sims.affectMotive("Hunger", heal_amt * 6)
+					H.sims.affectMotive("Bladder", -heal_amt * 0.2)
```
Comment (thegrb93) [inline code/modules/chemistry/tools/food_and_drink.dm:50]: Shouldn't this be healing instead of heal_amt? It was like this before though which is strange

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**